### PR TITLE
GEODE-8129: Added a drop proxy test.

### PR DESCRIPTION
**Pull request by mreddington**

### CODE REVIEW

**Review by mmartell — CHANGES_REQUESTED** [review]
No longer need the #if defined(_WIN32). SNI tests are enabled/disabled based on docker availability. The acceptance-tests will only run the linux platform in the CI. Please remove.

File: cppcache/acceptance-test/SNITest.cpp [hunk 1]
```diff
@@ -174,4 +174,51 @@ TEST_F(SNITest, connectWithoutProxyFails) {
...
+  auto region = cache.createRegionFactory(RegionShortcut::PROXY)
+                    .setPoolName("pool")
+                    .create("jellyfish");
+
+  auto systemRVal = std::system("docker-compose pause");
+  if (systemRVal == -1) {
+    BOOST_LOG_TRIVIAL(error) << "std::system returned: " << systemRVal;
+  }
+
+  region->put("1", "one");
```
Comment (echobravopapa) [inline cppcache/acceptance-test/SNITest.cpp:210]: you'll want to check/prove that the Put doesn't go through... i.e. its an expected failure and I bet there is an exception thrown...

**Review by echobravopapa — CHANGES_REQUESTED** [review]

File: cppcache/acceptance-test/SNITest.cpp (hunk 1 above)
Comment (mreddington) [reply cppcache/acceptance-test/SNITest.cpp:206]: That's literally impossible. A put or get will never block and never throw due to connectivity issues.

Comment (echobravopapa) [reply cppcache/acceptance-test/SNITest.cpp:206]: There won't be any exception or error after the connection times out?

**Pull request closed without merging**